src/pages/index.js

```diff
@@ -21,80 +21,271 @@ import {
+  })
+  .catch(() => (err) => console.log(err));
+//--------------------------------------------------------------AVATAR-------------------------------------------------------------------------------------
+//new avatar constants
+
+const profileAvatarModalImageSelector = document.querySelector(
+  // "#update-avatar-modal"
+  ".profile__image"
+);
+const profileAvatarModalSelector = document.querySelector(
+  "#update-avatar-modal"
+);
+const profileAvatarPeniclIcon = document.querySelector(
+  ".profile__icon-edit-button"
+);
+
+//avatar form validator instance
+
+const avatarFormValidator = new FormValidator(
+  settings,
+  profileAvatarModalSelector
+);
+avatarFormValidator.enableValidation();
+
+//edit avatar popup instance
+
+const editAvatarPopup = new PopupWithForm(
+  profileAvatarModalSelector,
+  handleAvatarFormSubmit
+);
+editAvatarPopup.setEventListeners();
+
+// avatar form submit function
+
+function handleAvatarFormSubmit({ url }) {
+  editAvatarPopup.setLoading(true);
+  api
+    .setUserAvatar(url)
+    .then((userData) => {
+      userInfo.setProfileAvatar(userData.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopup.setLoading(false, "Save");
+    });
 }
 
-//FORM VALIDATOR
+//calling avatar popup
 
-const addCardFormValidator = new FormValidator(settings, profileAddModal);
-addCardFormValidator.enableValidation();
+profileAvatarPeniclIcon.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-const editProfileFormValidator = new FormValidator(settings, profileEditModal);
-editProfileFormValidator.enableValidation();
+//--------------------------------------------------DELETE-----------------------------------------------------------------------------------
+// new delete card const
 
-//SECTION TO RENDER CARDS
+const deleteCardModalSelector = document.querySelector("#delete-confirm-modal");
 
-const cardListSection = new Section(
-  {
-    items: initialCards,
-    renderer: ({ name, link }) => {
-      const newCard = createCard({ name, link });
-      cardListSection.addItem(newCard);
-    },
-  },
-  cardList
+// delete card popup instance
+
+const deleteImagePopup = new PopupWithConfirm(
+  deleteCardModalSelector,
+  handleCardDelete
 );
+deleteImagePopup.setEventListeners();
 
-cardListSection.renderItems();
+// function for delete card
 
-// //POPUP WITH FORM
+function handleCardDelete() {
+  deleteImagePopup.setLoading(true);
+  const id = deleteImagePopup.getId();
+  api
 
-const newCardPopup = new PopupWithForm(profileAddModal, submitCard);
-newCardPopup.setEventListeners();
+    .deleteCard(id)
+    .then((res) => {
+      console.log(res);
+      deleteImagePopup.remove();
+      deleteImagePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      deleteImagePopup.setLoading(false, "Yes");
+    });
+  deleteImagePopup.open(id);
```

Comment:
`  deleteImagePopup.open(id);
`

Please, delete it. You don't need it in the submit handler